tests/nvmeof/test_ceph_nvmeof_neg_tests.py

```diff
@@ -5,154 +5,398 @@
+            nvme_service.deploy()
+
+        # Initialize gateways
+        nvme_service.init_gateways()
+        nvmegwcli = nvme_service.gateways[0]
+
+        # Configure Subsystems
+        configure_subsystems(nvme_service)
+
+        # Add listners
+        configure_listeners(nvme_service.gateways, nvme_service.config)
+
+        # Add host
+        configure_hosts(nvme_service.gateways[0], nvme_service.config)
+
+        # Create images
         name = generate_unique_id(length=4)
+        for i in range(5):
+            img = f"{name}-image{i}"
+            rbd.create_image(pool, img, "500M")
+            ns_args = {
+                "args": {
+                    "subsystem": config["initiators"][0]["subnqn"],
+                    "rbd-pool": pool,
+                    "rbd-image": img,
+                }
+            }
+            nvmegwcli.namespace.add(**ns_args)
+
+        @retry(IOError, tries=5, delay=5)
+        def list_subsystems(**sub_args):
+            try:
+                out, _ = nvmegwcli.subsystem.list(**sub_args)
+                return out
+            except Exception as e:
+                raise IOError(e)
+
+        sub_args = {"base_cmd_args": {"format": "json"}}
+        for initiator in config["initiators"]:
+            initiators(ceph_cluster, nvmegwcli, initiator)
+        gw_info_bkp = list_subsystems(**sub_args)
+        gw_info_bkp = json.loads(gw_info_bkp.strip())["subsystems"]
+
+        # restart nvmeof service
+        group = config.get("gw_group", "")
+        service_name = f"nvmeof.{pool}"
+        if group:
+            service_name = f"{service_name}.{group}"
+        restart_cfg = {
+            "no_cluster_state": False,
+            "config": {
+                "service": service_name,
+                "command": "restart",
+                "args": {"verify": True, "service_name": service_name},
+                "pos_args": [service_name],
+            },
+        }
+        test_orch.run(ceph_cluster, **restart_cfg)
+
+        gw_info = list_subsystems(**sub_args)
+        gw_info = json.loads(gw_info.strip())["subsystems"]
+
+        if gw_info != gw_info_bkp:
+            raise Exception(
+                f"GW Entities aren't same. Actual: {gw_info_bkp} Current: {gw_info}"
+            )
+        LOG.info("Validation of Ceph-83575467 is successful.")
+
+    except Exception as err:
+        raise Exception(err)
+
+
+def test_ceph_83575813(ceph_cluster, rbd, nvme_service, pool, config):
+    """CEPH-83575813: Perform NVMeoF RBD operations expand on images."""
+    client = get_node_by_id(ceph_cluster, config["initiator_node"])
+    initiator = NVMeInitiator(client)
+    try:
+        # Deploy NVMe Service
+        if config.get("install"):
+            nvme_service.deploy()
+
+        # Initialize gateways
+        nvme_service.init_gateways()
+        nvmegwcli = nvme_service.gateways[0]
+
+        # Configure Subsystems
+        configure_subsystems(nvme_service)
```

Comment:
Can we not directly call configure_gw_entities and that will take care of configuring subsystems, listeners and hosts right? All you need to do is pass those keywords to config I guess. Can we check this?